drivers/netdev2_ieee802154/netdev2_ieee802154.c

```diff
@@ -129,19 +106,13 @@ int netdev2_ieee802154_get(netdev2_ieee802154_t *dev, netopt_t opt, void *value,
             break;
 #ifdef MODULE_GNRC
         case NETOPT_PROTO:
-            if (max_len < sizeof(gnrc_nettype_t)) {
-                res = -EOVERFLOW;
-                break;
-            }
+            assert(max_len == sizeof(gnrc_nettype_t));
             *((gnrc_nettype_t *)value) = dev->proto;
             res = sizeof(gnrc_nettype_t);
             break;
 #endif
         case NETOPT_DEVICE_TYPE:
-            if (max_len < sizeof(uint16_t)) {
-                res = -EOVERFLOW;
-                break;
-            }
+            assert(max_len == sizeof(uint16_t));
```

Comment:
Question from above should have been stated here. I was wondering if (i) the assertion is needed and (ii) where to get the information about the size from. Of course the first thing I did was checking the netopt documentation but didn't find details for `NETOPT_DEVICE_TYPE` see [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/netopt.h#L178)